api/v3/Activity.php

```diff
@@ -372,6 +407,13 @@ function civicrm_api3_activity_get($params) {
       }
     }
   }
+
+  // Ensure there's enough data for calculating is_overdue
+  if (!empty($options['return']['is_overdue']) && (empty($options['return']['status_id']) || empty($options['return']['activity_date_time']))) {
+    $options['return']['status_id'] = $options['return']['activity_date_time'] = 1;
```

Comment:
I believe this will be considered as a bug to whoever notices this unusual behavior. People will probably waste some time debugging this just to find out that it was made "on purpose". Not to mention that I believe this will make the Activity API non-conformant (although I'm afraid it already is).

I'll leave this up to you. If you say it's not super important, than I think this PR is good to be merged